### ISSUE TITLE
custom attributes

### ISSUE THREAD

**Issue by blonkm** [FrozenNode/Laravel-Administrator#910]
How can I set the column of a relation to a custom attribute? Obviously a SELECT cannot find it in the table. I have a model that does not even have a name, it derives its name from other tables, so I created a custom attribute for it, like this:
```
    protected $appends = ['name', 'code', 'winner', 'loser', 'pointsA', 'pointsB'];

    public function getNameAttribute() {
            return $this->teamA->name . ' - ' . $this->teamB->name;
        }
```

**Comment by davemathews** [comment]
need more info, can you paste in your model and model config?

**Comment by judgej** [comment]
@blonkm is talking about [Accessors](http://laravel.com/docs/master/eloquent-mutators#accessors-and-mutators) not being usable in an index listing. I think it needs to be done through a custom rendering function. I've raised a separate ticket concerning making the underlying model available through the rendering callback so that this becomes much easier and with less duplication of code.

It is a tiny change to provide the renderer with the Eloquent model, as done in my fork here:

https://github.com/judgej/Laravel-Administrator/commit/e960d7df785fd68721b4f06a0ce1a3ab3e886c14

This was discussed on issue #775

**Comment by blonkm** [comment]
@judgej yes that is exactly what I am talking about. When I was reading about them, they were called *custom attributes*, not *accessors*. I know they're named accessors (or *access methods*) in Java, so it makes sense.

**Comment by thiagof** [comment]
Have you checked out `Using Accessors` from the [documentation](http://administrator.frozennode.com/docs/columns)?

**Comment by judgej** [comment]
@thiagof Yes, they are handy. I tend not to like putting stuff into models that is directly related to output and output formatting, so use them mostly just to get at raw data (it's about separation of concerns). It is good to know they are there as a fallback though.

**Comment by blonkm** [comment]
@judgej So I'm just interested, where do you put that code then? I haven't seen easy ways to make that reusable over several blade templates if it's not in the model (while maintaining Don't Repeat Yourself). Plus you're talking about this particular example (name), but 'winner', 'loser' and 'points' are attributes that need to be calculated, and thus have nothing to do with presentation. The database is certainly not the right place to put calculated attributes.

**Comment by judgej** [comment]
Looks like I left this one dangling :-) Getting back into a project that was on hold, so may be back with additional information.
